daemon/network.go

```diff
@@ -1120,7 +1120,10 @@ func buildJoinOptions(settings *network.Settings, n interface{ Name() string })
 		return []libnetwork.EndpointOption{}, nil
 	}
 
-	var joinOptions []libnetwork.EndpointOption
+	joinOptions := []libnetwork.EndpointOption{
+		libnetwork.JoinOptionPriority(epConfig.GwPriority),
```

Comment:
The hardest part reviewing this PR was to check "how does the sorting work?", and OHBOY things are libnetwork confusing; so

- the GwPriority is set on the endpoint config
- but set on the sandbox that it belongs to, indexed by endpoint-ID

https://github.com/moby/moby/blob/71b93a881cb765893c8c7d81a1c2e7b0c5623203/libnetwork/endpoint.go#L1247


So when does sorting happen?

This happens when an endpoint is added to a sandbox, it is injected in the right place in the list of endpoints attached to the sandbox https://github.com/moby/moby/blob/71b93a881cb765893c8c7d81a1c2e7b0c5623203/libnetwork/sandbox.go#L317-L320


:point_up: this is relevant, because it means that endpoints MUST go through that function to be added to the sandbox (just appending to the list won't put them in the right order)

But (prepare for a bumpy road here) :point_down:

The ordering happens by;

- each endpoint to get the sandbox it belongs to
- :question: as they're both either already connected, or _to be connected_ to the sandbox, wouldn't that sandbox always be the same???
- :point_right: even more fun; to get the sandbox;
    - the endpoint gets the network it belongs to
    - from which it gets the controller the network belongs to
    - which has a map of sandbox-ID -> sandbox
    - which is used to get the sandbox (which ... may be the sandbox we're coming from in the first place?)

https://github.com/moby/moby/blob/71b93a881cb765893c8c7d81a1c2e7b0c5623203/libnetwork/endpoint.go#L1086-L1097

After we have the sandbox, we can look up what priority was stored for the endpoint .. for that sandbox .. so that we can (if set) sort;
https://github.com/moby/moby/blob/71b93a881cb765893c8c7d81a1c2e7b0c5623203/libnetwork/sandbox.go#L664-L669

After which we do other things again, like (AGAIN) getting the network that the endpoint belongs to, to see if that network "internal" (which is also taken into account for priority);
https://github.com/moby/moby/blob/71b93a881cb765893c8c7d81a1c2e7b0c5623203/libnetwork/sandbox.go#L681-L683